### PR TITLE
refactor(CL): pool interfaces

### PR DESCRIPTION
**Pull request by p0mvn**
Closes: #3538

## What is the purpose of the change

Please note that the diff comes from regenerating pool mocks from new interfaces. Actual changes are quite small.

Refactoring pool interfaces:
- Renamed `TraditionalAmmInterface` to `CFMMI`
- Moved "Calc" and "Swap" methods from `PoolI` to `CFMMI`
- Created `convertToCFMMPool` to cast from `PoolI` to `CFMMI`
    * added tests
- Regenerated mocks from the new interfaces

### CODE REVIEW

File: x/gamm/keeper/pool_test.go [hunk 1]
```diff
@@ -316,3 +320,38 @@ func (suite *KeeperTestSuite) TestGetPoolAndPoke() {
...
+
+func (suite *KeeperTestSuite) TestConvertToCFMMPool() {
+	ctrl := gomock.NewController(suite.T())
+
+	tests := map[string]struct {
+		pool        swaproutertypes.PoolI
+		expectError bool
+	}{
+		"cfmm pool": {
+			pool: mocks.NewMockCFMMPoolI(ctrl),
```
Comment (czarcas7ic) [inline x/gamm/keeper/pool_test.go:332]: For my learning, whats the benefit of doing this over running a KeeperTestSuite helper func to make a pool?

File: x/swaprouter/simulation/sim_msgs.go [hunk 2]
```diff
@@ -152,10 +152,15 @@ func createPoolRestriction(k swaprouter.Keeper, sim *simtypes.SimCtx, ctx sdk.Co
 	}
 }
 
-func getRandPool(k simulationKeeper, sim *simtypes.SimCtx, ctx sdk.Context) (uint64, types.PoolI, sdk.Coin, sdk.Coin, []string, string, error) {
+// TODO: refactor this to work with simulateion pools
```
Comment (czarcas7ic) [inline x/swaprouter/simulation/sim_msgs.go:155]:
```suggestion
// TODO: refactor this to work with simulation pools
```

**Review by czarcas7ic — APPROVED** [review]
LGTM 👍

File: x/gamm/keeper/pool_test.go (hunk 1 above)
Comment (p0mvn) [reply x/gamm/keeper/pool_test.go:332]: Our helpers create a pool but don't return it. Instead, only return their id.

So we have to also call `GetPoolById()` to get the pool itself. It's more verbose.

Here, the goal is to simply test that given one pool implementation, the function passes. Given the other implementation, the function fails. So using mocks seemed to be the least verbose and simplest way to achieve that.

**Pull request merged**